### DISCUSSION TITLE
:Disk: displays my Btrfs subvolumes as individual partitions.

### DISCUSSION THREAD

**Discussion by Samueru-sama** [fastfetch-cli/fastfetch#395] [Q&A] [answered]
How can I prevent fastfetch from displaying every Btfs subvolume as an individual partition when using :Disk: in --Structure?

Currently it results in this: 

Disk (/): 15.73 GiB / 39.72 GiB (39%)
Disk (/.snapshots): 15.73 GiB / 39.72 GiB (39%)
Disk (/home): 15.73 GiB / 39.72 GiB (39%)
Disk (/mnt/GAMES): 101 GiB / 183 GiB (55%) [Removable]
Disk (/mnt/MEDIA): 245 GiB / 466 GiB (52%) [Removable]
Disk (/var/cache/pacman/pkg): 15.73 GiB / 39.72 GiB (39%)
Disk (/var/log): 15.73 GiB / 39.72 GiB (39%)

**Comment by LinusDierheimer** [comment]
Can you please post the content of your `/proc/mounts` file?

**Reply by Samueru-sama** [reply]
> Can you please post the content of your `/proc/mounts` file?

proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
sys /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
dev /dev devtmpfs rw,nosuid,relatime,size=4034616k,nr_inodes=1008654,mode=755,inode64 0 0
run /run tmpfs rw,nosuid,nodev,relatime,mode=755,inode64 0 0
efivarfs /sys/firmware/efi/efivars efivarfs rw,nosuid,nodev,noexec,relatime 0 0
/dev/sda2 / btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=292,subvol=/@ 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,nosuid,nodev,inode64 0 0
devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
cgroup2 /sys/fs/cgroup cgroup2 rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
bpf /sys/fs/bpf bpf rw,nosuid,nodev,noexec,relatime,mode=700 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=34,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=119 0 0
mqueue /dev/mqueue mqueue rw,nosuid,nodev,noexec,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,relatime,pagesize=2M 0 0
debugfs /sys/kernel/debug debugfs rw,nosuid,nodev,noexec,relatime 0 0
tracefs /sys/kernel/tracing tracefs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /tmp tmpfs rw,nosuid,nodev,size=4046004k,nr_inodes=1048576,inode64 0 0
fusectl /sys/fs/fuse/connections fusectl rw,nosuid,nodev,noexec,relatime 0 0
configfs /sys/kernel/config configfs rw,nosuid,nodev,noexec,relatime 0 0
ramfs /run/credentials/systemd-sysusers.service ramfs ro,nosuid,nodev,noexec,relatime,mode=700 0 0
ramfs /run/credentials/systemd-tmpfiles-setup-dev.service ramfs ro,nosuid,nodev,noexec,relatime,mode=700 0 0
/dev/sda2 /.snapshots btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=260,subvol=/@.snapshots 0 0
/dev/sda2 /home btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@home 0 0
/dev/sda2 /var/cache/pacman/pkg btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=259,subvol=/@pkg 0 0
/dev/sda2 /var/log btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=258,subvol=/@log 0 0
/dev/sda1 /boot vfat rw,noatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro 0 0
/dev/sda4 /mnt/GAMES btrfs rw,nosuid,nodev,noatime,compress-force=zstd:3,ssd,space_cache=v2,subvolid=5,subvol=/ 0 0
ramfs /run/credentials/systemd-tmpfiles-setup.service ramfs ro,nosuid,nodev,noexec,relatime,mode=700 0 0
ramfs /run/credentials/systemd-sysctl.service ramfs ro,nosuid,nodev,noexec,relatime,mode=700 0 0
/dev/sdb1 /mnt/MEDIA btrfs rw,nosuid,nodev,noatime,compress-force=zstd:3,space_cache=v2,subvolid=5,subvol=/ 0 0
tmpfs /run/user/1000 tmpfs rw,nosuid,nodev,relatime,size=809200k,nr_inodes=202300,mode=700,uid=1000,gid=1000,inode64 0 0
gvfsd-fuse /run/user/1000/gvfs fuse.gvfsd-fuse rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0

I do have 3 partitions in total my system (4 with the boot partition), those are:

Disk (/): 15.73 GiB / 39.72 GiB (39%)
Disk (/mnt/GAMES): 101 GiB / 183 GiB (55%) [Removable]
Disk (/mnt/MEDIA): 245 GiB / 466 GiB (52%) [Removable]

**Reply by LinusDierheimer** [reply]
What fastfetch currently parses is this:
```
/dev/sda2 /                     btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=292,subvol=/@ 0 0
/dev/sda2 /.snapshots           btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=260,subvol=/@.snapshots 0 0
/dev/sda2 /home                 btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@home 0 0
/dev/sda2 /var/cache/pacman/pkg btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=259,subvol=/@pkg 0 0
/dev/sda2 /var/log              btrfs rw,noatime,compress-force=zstd:3,ssd,discard=async,space_cache=v2,subvolid=258,subvol=/@log 0 0
/dev/sda1 /boot                 vfat  rw,noatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro 0 0
/dev/sda4 /mnt/GAMES            btrfs rw,nosuid,nodev,noatime,compress-force=zstd:3,ssd,space_cache=v2,subvolid=5,subvol=/ 0 0
/dev/sdb1 /mnt/MEDIA            btrfs rw,nosuid,nodev,noatime,compress-force=zstd:3,space_cache=v2,subvolid=5,subvol=/ 0 0
```
It detects everything starting with `/dev/` as device, `/boot` as hidden and everything with `nosuid,nodev` as removable. I am looking into a way to detect subvolumes.

**Comment by LinusDierheimer** [comment, marked as answer]
If you use the git version, they shouldn't show up anymore.

**Reply by Samueru-sama** [reply]
Hey, thank you for changing it, didn't think it would be something that required changes to the code.

**Reply by yuannan** [reply]
Is there a way to enable this? I actually quite like them all showing :joy:
